sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java

```diff
@@ -201,6 +219,14 @@ public AutoConfiguredOpenTelemetrySdk build() {
         LogEmitterProviderConfiguration.configureLogEmitterProvider(
             resource, config, meterProvider);
 
+    if (registerShutdownHook) {
+      Runtime.getRuntime().addShutdownHook(new Thread(tracerProvider::close));
```

Comment:
Let's just use one thread, and call shutdown with `CompletableResultCode.ofAll().join` to reduce thread explosion on shutdown